tests/bfd/bfd_helpers.py

```diff
@@ -158,41 +158,61 @@ def verify_static_route(
         )
 
 
-def control_interface_state(dut, asic, interface, action):
-    int_status = dut.show_interface(
-        command="status", include_internal_intfs=True, asic_index=asic.asic_index
-    )["ansible_facts"]["int_status"][interface]
-    oper_state = int_status["oper_state"]
+def batch_control_interface_state(dut, asic, interfaces, action):
     if action == "shutdown":
         target_state = "down"
     elif action == "startup":
         target_state = "up"
     else:
         raise ValueError("Invalid action specified")
 
-    if oper_state != target_state:
-        command = "shutdown" if action == "shutdown" else "startup"
-        exec_cmd = (
-            "sudo ip netns exec asic{} config interface -n asic{} {} {}".format(
-                asic.asic_index, asic.asic_index, command, interface
+    int_status = get_interfaces_status(dut, asic)
+    cmds = []
+    for interface in interfaces:
+        oper_state = int_status[interface]["oper_state"]
+        if oper_state != target_state:
+            command = "shutdown" if action == "shutdown" else "startup"
+            exec_cmd = "sudo ip netns exec asic{} config interface -n asic{} {} {}".format(
+                asic.asic_index,
+                asic.asic_index,
+                command,
+                interface,
             )
-        )
-        logger.info("Command: {}".format(exec_cmd))
-        logger.info("Target state: {}".format(target_state))
-        dut.shell(exec_cmd)
+
+            cmds.append(exec_cmd)
+            logger.info("Target state for interface {} is {}. Command: {}".format(
+                interface,
+                target_state,
+                exec_cmd,
+            ))
+        else:
+            raise ValueError("Invalid action specified for interface {}".format(interface))
+
+    if cmds:
+        dut.shell_cmds(cmds=cmds)
```

Comment:
I did a quick check on shell_cmds source code,
it actually runs the commands one by one: https://github.com/sonic-net/sonic-mgmt/blob/331254335e7455aa148f6d8c67d921e614afa4e4/ansible/library/shell_cmds.py#L137
So how much time we can expect to reduce with this refinement approach?

And how about we use multi-thread to run the interface commands? This way the commands will be executed in parallel

Reply:
We are seeing 20 seconds improvement on this as we only run one `dut.show_interface()` before the interfaces control command with this implementation. You are right, we can further improve this by running the startup/shutdown interfaces commands in parallel with mult-threading. Let me make some changes. Thanks!

Reply:
Hey @yejianquan, we are seeing another 45 seconds improvement after using multithreading to run those commands. So, in total we are seeing 65 seconds improvement for a single batch👍